sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -636,8 +666,9 @@ class reduction_impl : private reduction_impl_base {
   /// require initialization with identity value, then return user's read-write
   /// accessor. Otherwise, create 1-element global buffer initialized with
   /// identity value and return an accessor to that buffer.
+
   template <bool HasFastAtomics = has_fast_atomics>
-  std::enable_if_t<HasFastAtomics, rw_accessor_type>
+  std::enable_if_t<HasFastAtomics || has_atomic_add_float, rw_accessor_type>
```

Comment:
I'm not sure this matters, but it looks a little odd to me that `has_atomic_add_float` was added to the `enable_if` instead of to the template argument:
```c++
template <bool HasFastAtomics = (has_fast_atomics || has_atomic_add_float)>
```

Reply:
Yeah, that looks better.  I'll change it. Thanks

Reply:
I have now made this change as suggested.

Reply:
I have also renamed the has_atomic_add_float to has_atomic_add_float64, following our discussion in the next thread below.  I also renamed IsReduOptForAtomic64Add, which has_atomic_add_float64 is a pseudonym of.  This name was chosen to be more consistent with the naming convention of atomic64 specialization functions such as reduCGFuncAtomic64.